source/math/FloatEuclidTransform.ooc

```diff
@@ -43,6 +46,22 @@ FloatEuclidTransform: cover {
 		this rotation == other rotation &&
 		this scaling == other scaling
 	}
+	convolveCenter: static func (euclidTransforms: VectorList<This>, kernel: FloatVectorList) -> This {
+		result := This new()
+		if (euclidTransforms count > 0) {
+			result scaling = 0.0f
+			quaternions := VectorList<Quaternion> new(euclidTransforms count)
+			for (i in 0 .. euclidTransforms count) {
+				result translation = result translation + euclidTransforms[i] translation * kernel[i]
+				result scaling = result scaling + euclidTransforms[i] scaling * kernel[i]
+				rotation := euclidTransforms[i] rotation
+				quaternions add(rotation _quaternion)
```

Comment:
No idea, sorry :) 
It happened to me few times as well, when I wanted to create a complicated property like
```ooc
property ::= object member anotherMember something()
```
rock refused to compile. I remember having to create few more properties like
```ooc
tempProperty ::= object member 
anotherProperty ::= tempProperty anotherMember
property ::= anotherProperty something()
```
I've never actually took the time to debug it so I don't really know what's the reason behind this.